Fix latency using fixed-size threadpool instead of process spawn

cgu-tech opened the pull request:
* Rewrite process mechanism as a threadpool to minimize ressources consumption and latency (4 threads pool)
* Add websocket low-latency working client/server examples for remote mouse/trackpad

cgu-tech committed:
Fix latency using fixed-size threadpool instead of consuming process spawn (#1)

* Rewrite process mechanism as a threadpool to minimize ressources consumption and latency
* Add websocket low-latency working client/server samples

cgu-tech committed:
Add consumer keys capabilities (#2)

* HID gadget full rewrite into one device
* Now supports consumer keys (ex: AC_BACK for Android Back button)
* Lot of unit tests added

cgu-tech committed:
Add logs

cgu-tech committed:
Rewrite layouts and combos

cgu-tech committed:
Fix typo

cgu-tech committed:
Add combos detection

cgu-tech committed:
Fix default layout

thewh1teagle commented:
So sorry for the delay, I got so many notifications I completely lost track.
Really interesting solution. I’ll check it out over the weekend!

thewh1teagle commented:
Meantime quick reviews

Remove bash scripts for run
Replace python run commands/examples with uv (astral uv)

Examples comments should be docstring on top of the file:
"""
...
"""

code...

thewh1teagle commented:
I see that there's bunch of nice enhancements but the relevant changes for the fix could be few lines maybe 10
Could you create new PR only with the fix so we can address other enhancements later?

thewh1teagle commented:
It will be best if you tag me when commenting thanks 😅

cgu-tech committed:
Restore space char to all keymaps

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding microphone capabilities

cgu-tech committed:
Adding HID service robustness (#3)

See title

cgu-tech committed:
USB auto-reconnection part 1 (#4)

See title

cgu-tech committed:
USB auto-reconnection part 2 (#5)

Complete automatic USB recovery process (when HID is interrupted due to host, power outage, cable removed):
- part 1: automatically refresh HID file-descriptor on failed write attempt (one refresh per failed HID write attempt)
- part 2: automatically reload HID gadget at system level (one reload per two consecutive failed HID writes attempts)

cgu-tech committed:
Fix HID exception handling
